Tests/OTelCoreTests/Configuration/OTelConfigurationTests.swift

```diff
@@ -0,0 +1,763 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2025 the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import OTelCore
+import Testing
+
+@Suite struct ConfigurationTests {
+    // OTEL_SDK_DISABLED
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/
+    @Test(.disabled("Not currently implemented")) func testSDKDisabled() {}
+
+    // OTEL_LOG_LEVEL
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/
+    @Test func testDiagnosticsLogLevel() {
+        #expect(OTel.Configuration.default.diagnosticLogLevel.backing == .info)
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_LOG_LEVEL": "trace",
+        ]).diagnosticLogLevel.backing == .trace)
+    }
+
+    // OTEL_SERVICE_NAME
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/
+    // https://opentelemetry.io/docs/languages/sdk-configuration/general/#otel_service_name
+    @Test func testServiceName() {
+        #expect(OTel.Configuration.default.serviceName == "unknown_service")
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_SERVICE_NAME": "some_service",
+        ]).serviceName == "some_service")
+
+        withKnownIssue {
+            Issue.record("""
+            This property is a special since it's supposed to affect what ends up as the value for the service.name
+            resource attribute during export.
+
+            However, I don't think the right way to do that is to magically change the resource attributes of the confg
+            object that the user has.
+
+            Most likely this should be solved with a computed property on the config that's used by exporters, e.g.
+            `resolvedResourceAttributes`.
```

Comment:
```suggestion
            This property is special since it's supposed to affect what ends up as the value for the service.name
            resource attribute during export.

            However, I don't think the right way to do that is to magically change the resource attributes of the config
            object that the user has.

            Most likely this should be solved with a computed property on the config that's used by exporters, e.g.
            `resolvedResourceAttributes`.
```

I agree that this should likely not mutate the configuration's resource.